build/android/src/main/java/net.minetest.minetest/MinetestAssetCopy.java

```diff
@@ -1,416 +1,373 @@
 import android.os.Environment;
 import android.util.Log;
 import android.view.Display;
+import android.view.View;
 import android.widget.ProgressBar;
 import android.widget.TextView;
-import android.graphics.Rect;
-import android.graphics.Paint;
-import android.text.TextPaint;
 
-public class MinetestAssetCopy extends Activity
-{
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.util.Iterator;
+import java.util.Vector;
+
+public class MinetestAssetCopy extends Activity {
+	ProgressBar m_ProgressBar;
+	TextView m_Filename;
+	copyAssetTask m_AssetCopy;
+
 	@Override
-	public void onCreate(Bundle savedInstanceState)
-	{
+	public void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
-		
 		setContentView(R.layout.assetcopy);
-		
-		m_ProgressBar = (ProgressBar) findViewById(R.id.progressBar1);
-		m_Filename = (TextView) findViewById(R.id.textView1);
-		
+		m_ProgressBar = findViewById(R.id.progressBar1);
+		m_Filename = findViewById(R.id.textView1);
 		Display display = getWindowManager().getDefaultDisplay();
 		m_ProgressBar.getLayoutParams().width = (int) (display.getWidth() * 0.8);
 		m_ProgressBar.invalidate();
-		
+
 		/* check if there's already a copy in progress and reuse in case it is*/
-		MinetestAssetCopy prevActivity = 
+		MinetestAssetCopy prevActivity =
 				(MinetestAssetCopy) getLastNonConfigurationInstance();
-		if(prevActivity!= null) {
+		if (prevActivity != null) {
 			m_AssetCopy = prevActivity.m_AssetCopy;
-		}
-		else {
+		} else {
 			m_AssetCopy = new copyAssetTask();
 			m_AssetCopy.execute();
 		}
 	}
-	
+
+	@Override
+	protected void onResume() {
+		super.onResume();
+		makeFullScreen();
+	}
+
+	public void makeFullScreen() {
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+			this.getWindow().getDecorView().setSystemUiVisibility(
+					View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
+			);
+		}
+	}
+
+	@Override
+	public void onWindowFocusChanged(boolean hasFocus) {
+		super.onWindowFocusChanged(hasFocus);
+		if (hasFocus) {
+			makeFullScreen();
+		}
+	}
+
 	/* preserve asset copy background task to prevent restart of copying */
 	/* this way of doing it is not recommended for latest android version */
 	/* but the recommended way isn't available on android 2.x */
-	public Object onRetainNonConfigurationInstance()
-	{
+	public Object onRetainNonConfigurationInstance() {
 		return this;
 	}
-	
-	ProgressBar m_ProgressBar;
-	TextView m_Filename;
-	
-	copyAssetTask m_AssetCopy;
-	
-	private class copyAssetTask extends AsyncTask<String, Integer, String>
-	{
-		private long getFullSize(String filename)
-		{
+
+	private class copyAssetTask extends AsyncTask<String, Integer, String> {
+		boolean m_copy_started = false;
```

Comment:
`m_` is rarely used in Java (and is discouraged in future code in our code base)

Reply:
same